File: data-plane/gateway/datapath/src/commands.rs
// Copyright AGNTCY Contributors (https://github.com/agntcy)
// SPDX-License-Identifier: Apache-2.0

use tokio::sync::oneshot;

#[derive(Debug)]
pub enum ControlCommand {
    Subscribe {
        reply: oneshot::Sender<Result<(), String>>,
    },
    Unsubscribe {
        reply: oneshot::Sender<Result<(), String>>,
    },
}

File: control-plane/internal/controller/commands.go
// Copyright AGNTCY Contributors (https://github.com/agntcy)
// SPDX-License-Identifier: Apache-2.0

package controller

type ControlCommand interface {
	// CommandType returns the type (subscribe, unsubscribe, etc.)
	CommandType() ControlCommandType
}

type ControlCommandType int

const (
	Subscribe ControlCommandType = iota
	Unsubscribe
)

type SubscribeCommand struct {
}

func (sc SubscribeCommand) CommandType() ControlCommandType {
	return Subscribe
}

type UnsubscribeCommand struct {
}

func (uc UnsubscribeCommand) CommandType() ControlCommandType {
	return Unsubscribe
}
